core/lib.rs

```diff
@@ -1114,8 +1114,24 @@ impl Statement {
         if res.is_err() {
             let state = self.program.connection.transaction_state.get();
```

Comment:
We do a very similar thing here as in `self.program.step`. Can we have sth like

```rust
    pub fn step(&mut self) -> Result<StepResult> {
        self.handle_err(self.program.step(...))
    }
    
    fn handle_err<T>(&self, res: Result<T>) -> Result<T> {
      if res.is_err() {
         // pager rollback etc
      }
    } 
    
    pub fn run_once(&self) -> Result<()> {
       self.handle_err(self.pager.io.run_once())
    }
    ```
    
    and then not have the very similar error handling in `Program::step`?